### DISCUSSION TITLE
[ 草案规划 ] 关于跨端生态的想法

### DISCUSSION THREAD

**Discussion by SOVLOOKUP** [rubickCenter/rubick#58] [Ideas]
## Ideas
跨端 = 跨移动 + PC端，从utools微信群和[社区的用户反馈](https://yuanliao.info/d/2316-utools)来看，用户对移动端有迫切的需求。

从统计来看，全球70%的流量终端都是移动设备，且以后这个数量还在继续上涨，大多数用户在接触PC前被培养的用户习惯都是来源于移动平台。

从Google的Chrome OS、华为的HarmonyOS、微软的UWP等战略产品也可以看出未来多端统一的趋势。

---

移动端有一款工具箱叫 [一个木函](https://www.woobx.cn/) 小米商城的下载量有1400多万、华为app商城下载量有1700多万，拥有非常庞大的用户流量，如果能辐射到桌面端，就会有更强大的流量变现能力。

而这个移动端的工具箱因为历史代码原因，无法适应桌面端，同样，utools因为历史代码的原因，无法快速开发出移动端。

但我们能从顶层就开始思考，如何为用户实现原生 跨端 + 跨平台 的体验，创新会让我们从同类中脱颖而出。

## Tech

技术方案上 uniapp 已经实现了移动端的跨平台，所以有一种实现全生态技术方案的可能性是：

将uniapp的移动端跨平台能力和electron的桌面端跨平台能力结合

## 参考资料

uniapp跨端开发：[在开发中可以调用electron的API](https://uniapp.dcloud.net.cn/adapt?id=%e9%80%9a%e8%bf%87electron%e6%89%93%e5%8c%85%e4%b8%bawindows%e3%80%81mac%e3%80%81linux%e5%ae%a2%e6%88%b7%e7%ab%af)

**Comment by muwoo** [comment]
可能是个不错的方向

**Comment by euukey** [comment]
看好uniapp

**Comment by KxiaXX** [comment]
非常看好，加油！

**Comment by SOVLOOKUP** [comment]
在对可行性进行深入研究、并测试了几个demo以后，很遗憾，uniapp其实并不符合本项目对于硬件兼容层的要求。

由于uniapp更偏向于一个移动端的偏视图层的框架，无法满足本项目UI及底层能力的定制需求。

不过跨端仍然是可行的，[tauri](https://github.com/tauri-apps/tauri) 这个项目已经有基本取代electron的能力，并对ios和安卓进行支持，对node也会支持，这将是一个很好的electron替代方案和跨端发展方向 

感兴趣的朋友可以一起持续关注这个方向
